### PR TITLE
Update Errors in Mqtt Client 

### PR DESCRIPTION
**Pull request by xiazhvera**
*Issue #, if available:*

*Description of changes:*
1. Added error space for aws-crt-swift
2. Added a way to allow customized error message for Swift
3. Added `unwrap` function to unwrap Optional

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (sfod) [inline Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift:28]: Is it intentional that `code == 0` is treated as unknown error? It doesn't feel right, because `AWS_ERROR_SUCCESS` is valid code and can be passed to this function, even if by accident (e.g. by `aws_last_error ()` call).

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (sbSteveK) [inline Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift:26]: Do we still need the context string if we are generating errors from crt swift?

In the areas that are currently using context, it seems like error logging would be more appropriate. Unless we are not going to generate logs from aws-crt-swift.

**Review by sbSteveK — APPROVED** [review]
Unsure whether we need the context string in CommonRuntimeError. That can be changed later if we decide against it. Looks solid otherwise.

On a side note, once we do this, we need to update common to reflect the error range that aws-crt-swift is taking.

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (waahm7) [inline Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift:26]: We should just update the existing function. It won't break anything.

File: Source/AwsCommonRuntimeKit/crt/Utilities.swift
Comment (waahm7) [inline Source/AwsCommonRuntimeKit/crt/Utilities.swift:151]: We should avoid duplicating errors because they are "common". The distinction between AWS_ERROR_INVALID_ARGUMENT and AWS_CRT_SWIFT_INVALID_ARGUMENT might not be clear to customers, leading to confusion. Having two different errors for the same issue can be unnecessarily complicated.

File: Source/AwsCommonRuntimeKit/crt/Utilities.swift
Comment (waahm7) [inline Source/AwsCommonRuntimeKit/crt/Utilities.swift:233]: This feels like something Swift standard API should already provide. How is this different from `.?`?

File: Source/LibNative/module.modulemap
Comment (waahm7) [inline Source/LibNative/module.modulemap:6]: Why not put it in https://github.com/awslabs/aws-crt-swift/tree/main/aws-common-runtime/config?

File: Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift
Comment (waahm7) [inline Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift:289]: Debatable: Probably better to just use `AWS_ERROR_INVALID_STATE` with context?

**Review by waahm7 — COMMENTED** [review]
Added comments. I am willing to agree to disagree here. I still feel that this PR doesn't have a strong enough use case for Swift errors. The added complexity might not justify the benefits, which seem limited to a slightly better error name that might not be significant for users. Our error handling and Swift error handling is terrible, and I understand the desire to make it better for the customers. However, this just feels like a bandage which doesn't solve the issue. I would suggest holding off on merging this until we identify a more concrete use case, but I'll defer to your judgment.

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (waahm7) [inline Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift:17]: I think this was intentional. This can lead a bad call to last error becoming AWS_ERROR_SUCCESS, which will probably make the program fail at some later stage, making it harder to debug. I think it is easier to debug where the unknown error is coming from and fix that.

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (waahm7) [reply Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift]: Yes, that was intentional. This can lead a bad call to last error becoming AWS_ERROR_SUCCESS, which will probably make the program fail at some later stage, making it harder to debug. I think it is easier to debug where the unknown error is coming from and fix that.

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (waahm7) [reply Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift:17]: Or customers will get exceptions with AWS_ERROR_SUCCESS instead of ERROR_UNKNOWN

File: Source/LibNative/module.modulemap
Comment (xiazhvera) [reply Source/LibNative/module.modulemap:6]: The `config` module is required to build before `aws-c-common` because it is used for build configuration. Meanwhile, the `error` file uses error struct from `aws-c-common`, which introduces a dependency on `aws-c-common`. If we add the error file into config module, it would be a recursive dependency which is not allowed by the swift package.

File: Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift
Comment (xiazhvera) [reply Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift:289]: I decided to keep the Swift error space, as it makes sense to maintain it whether we use it or not.  Consequently, I am also keeping the `AWS_CRT_SWIFT_MQTT_CLIENT_CLOSED` error code. 
Additionally, I removed `AWS_CRT_SWIFT_INVALID_ARGUMENT ` to avoid confusion with `AWS_ERROR_INVALID_ARGUMENT `

File: Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift
Comment (xiazhvera) [reply Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift]: I eventually decide to keep the context, as the logging for Swift is not there yet.

**Pull request merged**
